sql/lesson5/tutorial.md

```diff
@@ -0,0 +1,181 @@
+---
+layout: page
+title: Introduction to SQL
+---
+
+## About Lesson 5
+
+The previous lessons showed you how to manipulate data from a single table. _Lesson 5_ will show you how to combine data from multiple tables and will bring out the _relational_ part of relational databases.
+
+Tables in a relational database relate to each other as they contain keys which allow information from multiple tables to be combined easily. The real power of relational databases, and SQL, comes from working with multiple tables at once.
+
+In this lesson we will keep using the same database as before (https://sqliteonline.com/#fiddle-5be888fa749e451ljodase3y)
+
+### Primary and Foreign Keys
+
+All tables in a relational database need to have a primary key. This is an identifier that uniquely identifies each row. We saw in _Lesson 4_ that the `people` table uses the `id` field as a primary key. The key aspects of the primary key field are that it is never empty and it is unique: every record in a table has one and only one record can have a given primary key. For instance, in the `people` table, record with `id` `1` is the record for `Emma Smith`. If we try to insert a new record with the same `id` primary key, it will fail. If we try to insert a record with no value for `id` it will fail too.
+
+Any table can also have a number of foreign key fields. Each foreign key contains the value for a primary key in another table. For example, if we select the data in the `hosts` table:
+
+```SQL
+SELECT *
+  FROM hosts;
+```
+
+You will notice the `location_id` field. This field is a foreign key to the `locations` table. The value of this field for the first record, Happy Code is `2`. So let's have a look at the locations table:
+
+```SQL
+SELECT *
+  FROM locations;
+```
+
+This shows us that the record in the `locations` table that has `id` set to `2` is the location record for Brighton in the UK. That's great, we can now go from `hosts` to `locations` with two queries but can we get this data in a single query and for all records? This is where table joins come in.
+
+### Inner join
+
+#### Joining two tables
```

Comment:
Suggest in this section you dissect the statement syntax to make it clear what the ON part is about, whereupon you can refer to it as specifying the _reference_ (primary key and foreign key), then it ties up a bit better with the explanation above.

Reply:
This may also help them to disambiguate the syntax with a WHERE clause (and we'll just keep quiet about how it essentially _IS_ a where clause :D )

Reply:
I'll add the following dissection:

An INNER JOIN query can be generalized as follows:
```SQL
SELECT {columns}
  FROM {table_A}
 INNER JOIN {table_B}
    ON {table_B.primary_key} = {table_A.foreign_key_to_B}
```
In the queries above, the `.` is used to indicate to which table does the field correspond to. This is known as _dot notation_. When any of the fields in your statement can be found in more than one of the table involved, you must use this notation to make it clear to which one you are referring to.